packages/react/src/ReactLazy.js

```diff
@@ -53,29 +53,17 @@ function lazyInitializer<T>(payload: Payload<T>): T {
     const ctor = payload._result;
     const thenable = ctor();
     // Transition to the next state.
-    const pending: PendingPayload = (payload: any);
-    pending._status = Pending;
```

Comment:
What if it was a sync thenable though? Wouldn't you miss it due to `if (payload._status === Pending) {` check in `onFulfill`?

Reply:
Yea I fixed it here. https://github.com/facebook/react/pull/21642/commits/cc42e7b0fc429ce9ed8a5b5f7340d1ae0b483309

That check is really unnecessary. It's more of a safety thing in case someone has a bad thennable. We could probably remove it.